### PR TITLE
fix(userspace/falco): properly format numeric values in metrics

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind bug

/kind cleanup

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:

Following up #2333, this makes sure that numeric values are formatted as numbers in the JSON outputs of the new metrics system.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

cc @incertum

**Does this PR introduce a user-facing change?**:

```release-note
fix(userspace/falco): properly format numeric values in metrics
```

### CODE REVIEW

**Review by leogr — DISMISSED** [review]
I don't like too much using `nlohmann::json`, but I can't think of a better solution right now.

So, LGTM.

**Comment by jasondellaluce** [comment]
> I don't like too much using nlohmann::json, but I can't think of a better solution right now.

Same 😸

**Review by leogr — DISMISSED** [review]

File: userspace/falco/stats_writer.cpp
Comment (FedeDP) [inline userspace/falco/stats_writer.cpp:1]: @jasondellaluce 
please consider fixing https://github.com/falcosecurity/falco/pull/2333#discussion_r1201717285 here :)

**Review by FedeDP — COMMENTED** [review]
LGTM; left a comment about `strncat` issue.
/hold

**Comment by jasondellaluce** [comment]
> Wouldn't we need a ! as we change the data types for all Falco rules not just the internal stats / metrics related rules?

I don't think so, this changes the types only for the new metrics internal alerts, which is something we haven't released yet.

**Comment by incertum** [comment]
> > Wouldn't we need a ! as we change the data types for all Falco rules not just the internal stats / metrics related rules?
> 
> I don't think so, this changes the types only for the new metrics internal alerts, which is something we haven't released yet.

Wait so `handle_msg` is just used for the internal rules and not for the regular Falco rules? In that case just out of curiosity could you point me to the place where regular Falco rules are formatted?

**Review by leogr — APPROVED** [review]

**Pull request merged**